### PR TITLE
Karmadactl init: optimize getKarmadaAPIServerIP

### PR DESCRIPTION
**Pull request by helen-frank**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
getKarmadaAPIServerIP do not need to list node multiple times
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Speed up for init
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by helen-frank** [comment]
@carlory @lonelyCZ cc

File: pkg/karmadactl/cmdinit/kubernetes/node.go [hunk 1]
```diff
@@ -26,43 +26,34 @@ func (i *CommandInitOption) getKarmadaAPIServerIP() error {
...
 	klog.Info("randomly select 3 Node IPs in the kubernetes cluster.")
-	nodes, err := nodeClient.List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-
 	for number := 0; number < 3; number++ {
 		if number >= len(nodes.Items) {
-			break
+			if len(i.KarmadaAPIServerIP) == 0 {
```
Comment (lonelyCZ) [inline pkg/karmadactl/cmdinit/kubernetes/node.go:50]: This code seems not necessary to embed here. Putting it the end could be clearer.

Comment (helen-frank) [reply pkg/karmadactl/cmdinit/kubernetes/node.go]: I apologize for not replying to your message for such a long time (I am infected with COVID-19 and have just improved recently😀)

I adjusted the original code logic

Comment (RainbowMango) [reply pkg/karmadactl/cmdinit/kubernetes/node.go]: :) I'm infected with COVID-19 as well...

**Comment by RainbowMango** [comment]
Just a reminder: This branch has conflicts that must be resolved.

**Comment by RainbowMango** [comment]
Thanks for your quick response.
/assign

**Comment by RainbowMango** [comment]
I'm sorry for letting this sit so long. After going through the code, listing nodes more than once is probably not a problem. 

Since in most cases, a cluster would have the nodes with `node-role.kubernetes.io/control-plane` (or `node-role.kubernetes.io/master`), listing with a label selector could limit the requested data. It would be helpful for clusters with a large number of nodes.

In other words, we usually don't need to list all nodes.

**Comment by helen-frank** [comment]
> I'm sorry for letting this sit so long. After going through the code, listing nodes more than once is probably not a problem.
> 
> Since in most cases, a cluster would have the nodes with `node-role.kubernetes.io/control-plane` (or `node-role.kubernetes.io/master`), listing with a label selector could limit the requested data. It would be helpful for clusters with a large number of nodes.
> 
> In other words, we usually don't need to list all nodes.

This is indeed the case. My more consideration is that I can take out all the nodes at once and then filter the nodes of `node-role.kubernetes.io/control-plane` or `node-role.kubernetes.io/master`. You can consider this angle. If it is not acceptable, I will close the pr.

**Comment by RainbowMango** [comment]
> You can consider this angle. 

Yeah, I looked at the patch. And my concern is that maybe we usually don't need to fetch all nodes. You know, node configuration is pretty large.

**Comment by helen-frank** [comment]
got it and /close

**Pull request closed without merging**